tests/ProxyManagerTest/Functional/LazyLoadingGhostFunctionalTest.php

```diff
@@ -1070,4 +1073,154 @@ public function skipPropertiesFixture()
+                $initializer = null;
+
+                $props[$propertyIndex] = $expectedValue;
+            }
+        );
+
+        /* @var $accessor callable */
+        $accessor = [$callerObject, $method];
+
+        self::assertFalse($proxy->isProxyInitialized());
+        self::assertSame($expectedValue, $accessor($proxy));
+        self::assertTrue($proxy->isProxyInitialized());
+    }
+
+    /**
+     * @group 276
+     *
+     * @dataProvider getMethodsThatAccessPropertiesOnOtherObjectsInTheSameScope
+     *
+     * @param object $callerObject
+     * @param string $method
+     * @param string $propertyIndex
+     * @param string $expectedValue
+     */
+    public function testWillAccessMembersOfOtherDeSerializedProxiesWithTheSamePrivateScope(
+        $callerObject,
+        string $method,
+        string $propertyIndex,
+        string $expectedValue
+    ) {
+        $proxyName = $this->generateProxy(get_class($callerObject));
+        /* @var $proxy OtherObjectAccessClass|LazyLoadingInterface */
+        $proxy = unserialize(serialize($proxyName::staticProxyConstructor(
+            function ($proxy, $method, $params, & $initializer, array $props) use ($propertyIndex, $expectedValue) {
+                $initializer = null;
+
+                $props[$propertyIndex] = $expectedValue;
+            }
+        )));
+
+        /* @var $accessor callable */
+        $accessor = [$callerObject, $method];
+
+        self::assertTrue($proxy->isProxyInitialized());
+        self::assertSame($expectedValue, $accessor($proxy));
+    }
+
+    /**
+     * @group 276
+     *
+     * @dataProvider getMethodsThatAccessPropertiesOnOtherObjectsInTheSameScope
+     *
+     * @param object $callerObject
+     * @param string $method
+     * @param string $propertyIndex
+     * @param string $expectedValue
+     */
+    public function testWillAccessMembersOfOtherClonedProxiesWithTheSamePrivateScope(
+        $callerObject,
+        string $method,
+        string $propertyIndex,
+        string $expectedValue
+    ) {
+        $proxyName = $this->generateProxy(get_class($callerObject));
+        /* @var $proxy OtherObjectAccessClass|LazyLoadingInterface */
+        $proxy = clone $proxyName::staticProxyConstructor(
+            function ($proxy, $method, $params, & $initializer, array $props) use ($propertyIndex, $expectedValue) {
+                $initializer = null;
+
+                $props[$propertyIndex] = $expectedValue;
+            }
+        );
+
+        /* @var $accessor callable */
+        $accessor = [$callerObject, $method];
+
+        self::assertTrue($proxy->isProxyInitialized());
+        self::assertSame($expectedValue, $accessor($proxy));
+    }
+
+    public function getMethodsThatAccessPropertiesOnOtherObjectsInTheSameScope() : array
+    {
+        $proxyClass = $this->generateProxy(OtherObjectAccessClass::class);
+
+        return [
+            OtherObjectAccessClass::class . '#$privateProperty' => [
```

Comment:
I'm mainly doing this for readability/later refactoring. It gets really complicated to look for `OtherObjectAccessClass` usages otherwise...